src/main/scala/Main.scala

```diff
@@ -1,3 +1,103 @@
 object Main extends App {
-  println("Hello, world!")
+  // TASK 1
+  def max(arr: Array[Int]): Option[Int] = if (arr.isEmpty) None else Some(arr.reduce((x, y) => x max y))
+
+  def min(arr: Array[Int]): Option[Int] = if (arr.isEmpty) None else Some(arr.reduce((x, y) => x min y))
+
+  def sum(arr: Array[Int]): Int = arr.fold(0)((x, y) => x + y)
+
+
+  /**
+   * Returns a card's value as an integer.
+   * Returns None if the string passed is not a valid card.
+   *
+   * @param card The card representation as string
+   * @return The card's value
+   */
+  def parse(card: String): Option[Int] = {
+    card match {
+      case "A" => Some(11)
+      case "J" | "Q" | "K" => Some(10)
+      case _ =>
+        try {
+          val number = card.toInt
```

Comment:
Great use of `Option`!

A more idiomatic way to express
```scala
case _ =>
        try {
          val number = card.toInt
          if (number < 2 || number > 10) None else Some(number)
        } catch {
          case _: NumberFormatException => None
        }
```
would be
```scala
case _ => Try(card.toInt).toOption.flatMap(x => if (x < 2 || x > 10) None else Some(x))
```
We will look into that in future lectures